.github/workflows/release.yaml

```diff
@@ -57,18 +57,42 @@ jobs:
           use-cross: ${{ matrix.job.use-cross }}
           command: build
           args: --release --target ${{ matrix.job.target }}
+
+      - name: Strip release binary (linux and macOS)
+        if: matrix.job.os != 'windows-latest'
+        run: |
+          if [ "${{ matrix.job.target }}" = "arm-unknown-linux-gnueabihf" ]; then
+            sudo apt-get -y install gcc-arm-linux-gnueabihf
+            arm-linux-gnueabihf-strip "target/${{ matrix.job.target }}/release/xh"
+          else
+            strip "target/${{ matrix.job.target }}/release/xh"
+          fi
+
+      - id: get_version
+        uses: battila7/get-version-action@v2
       - name: Package
         shell: bash
         run: |
-          cd target/${{ matrix.job.target }}/release
           if [ "${{ matrix.job.os }}" = "windows-latest" ]; then
-            7z a ../../../xh-${{ matrix.job.target }}.zip xh.exe
+            bin="target/${{ matrix.job.target }}/release/xh.exe"
+          else
+            bin="target/${{ matrix.job.target }}/release/xh"
+          fi
+          staging="xh-${{ steps.get_version.outputs.version }}-${{ matrix.job.target }}"
+
+          mkdir -p "$staging"/{doc,completions}
+          cp LICENSE README.md $bin $staging
+          cp CHANGELOG.md doc/xh.1 "$staging"/doc
+          cp completions/* "$staging"/completions
+
+          if [ "${{ matrix.job.os }}" = "windows-latest" ]; then
+            7z a "$staging.zip" $staging
           elif [ "${{ matrix.job.os }}" = "macos-latest" ]; then
-            zip ../../../xh-${{ matrix.job.target }}.zip xh
+            zip -r "$staging.zip" $staging
```

Comment:
In case anyone wonders why not use the `tar` command for macOS, it is because it was producing [corrupted output](https://github.com/actions/upload-artifact/issues/151).

Reply:
GNU tar should be installed by default now, so maybe we can just run `gtar`?

Reply:
It seems `gtar` doesn't produce corrupted artifacts 👍